editor/js/components/arcontroller.component.js

```diff
@@ -71,6 +71,55 @@ ArControllerComponent.prototype.onRemovedFromScene = function( scene ) {
 ArControllerComponent.prototype.startAR = function() {
     console.log("Start AR");
     var self = null;
+    var scale = 0.05;
+    var mesh = GL.Mesh.plane({xz: true,width: parseInt(gl.canvas.width*scale),height:parseInt(gl.canvas.height*scale),detail:0});
+    var type = gl.HIGH_PRECISION_FORMAT;
+
+    var texture = undefined;
+    //basic shader
+    //create basic matrices for cameras and transformation
+    var proj = mat4.create();
+    var view = mat4.create();
+    var model = mat4.create();
+    var mvp = mat4.create();
+    var temp = mat4.create();
+
+    //get mouse actions
+                  //set the camera position
+    mat4.perspective(proj, 45* DEG2RAD, gl.canvas.width / gl.canvas.height, this.nearPlane, this.farPlane);
+    mat4.lookAt(view, [0,30,30],[0,0,0], [0,1,0]);
+
+    //basic phong shader
+    var shader = new Shader('\
+        precision highp float;\
+        attribute vec3 a_vertex;\
+        attribute vec3 a_normal;\
+        attribute vec2 a_coord;\
+        varying vec3 v_normal;\
+        varying vec2 v_coord;\
+        uniform mat4 u_mvp;\
+        uniform mat4 u_model;\
+        void main() {\
+          v_coord = a_coord;\
+          v_normal = (u_model * vec4(a_normal,1.0)).xyz;\
+          gl_Position = u_mvp * vec4(a_vertex,1.0);\
+        }\
+        ', '\
+        precision highp float;\
+        varying vec3 v_normal;\
+        varying vec2 v_coord;\
+        uniform vec3 u_lightvector;\
+        uniform vec4 u_color;\
+        uniform sampler2D u_texture;\
+        void main() {\
+          vec3 N = normalize(v_normal);\
+          vec4 color = u_color * texture2D( u_texture, v_coord);\
```

Comment:
If you make this change you will see complitely the video source. I think the shader should be different, that you can see the video stream and the cube onto it.
```diff
- gl_FragColor = color * max(0.5, dot(u_lightvector,N));\
+ gl_FragColor = color * max(1.0, dot(u_lightvector,N));\

```

Reply:
@kalwalt 
You are right, the problem is howto render right way using shader, since I used litegl, we have to check howto do this with litegl or ask to someone (shader just takes video pixel to render it as texture).
@ThorstenBux  sorry I didn't try to resize, resize function must be implemented to resize plane or move camera to fit to the video plane. The idea is to do all resize inside de 3d world, moving camera or resizeing plane, the canvas takes need size.

Reply:
I fixed the problem of the shader ... Tomorrow If I had time I would like to implemented resizing and other bugs ;)
![Screenshot from 2020-03-17 21-47-23](https://user-images.githubusercontent.com/3738104/76900051-eedd0000-6898-11ea-9237-372783b81dae.png)

Reply:
Well done! Setting position of the cube seems not having effect, don't understand why...